examples/jsm/loaders/GLTFLoader.js

```diff
@@ -3524,17 +3524,17 @@ class GLTFParser {
 
 		const sanitizedName = PropertyBinding.sanitizeNodeName( originalName || '' );
 
-		if ( sanitizedName in this.nodeNamesUsed ) {
+		let uniqueName = sanitizedName;
 
-			return sanitizedName + '_' + ( ++ this.nodeNamesUsed[ sanitizedName ] );
+		if ( sanitizedName in this.nodeNamesUsed ) {
 
-		} else {
+			uniqueName = sanitizedName + '_' + ( ++ this.nodeNamesUsed[ sanitizedName ] );
 
-			this.nodeNamesUsed[ sanitizedName ] = 0;
+		}
 
-			return sanitizedName;
+		this.nodeNamesUsed[ uniqueName ] = 0;
 
-		}
+		return uniqueName;
```

Comment:
I feel like I am missing something obvious sorry, but I can't see what this does differently than the previous code — what was the problem and solution here?

Reply:
Thank you for the question.

The goal of this change is to ensure the uniqueness of node name. The previous implementation could return non-unique node name on a certain case. The issue is the array this.nodeNamesUsed doesn't memorize all of the node values because of early return on line 3529.

Here's an example, given two nodes:
- First node with name "triangle" contains two primitives, each without a name.
- Second node with name "triangle_1" contains one primitive without a name.

The loader will then process each primitive as follows:
In the old implementation:
```
[1] sanitizedName = "triangle"
    sanitizedName in this.nodeNamesUsed = False
    this.nodeNamesUsed["triangle"] = 0
    > return "triangle" 

[2] sanitizedName = "triangle"
    sanitizedName in this.nodeNamesUsed = True
    this.nodeNamesUsed["triangle"] = 1
    > return "triangle_1"

[3] sanitizedName = "triangle_1"
    sanitizedName in this.nodeNamesUsed = False
    this.nodeNamesUsed["triangle_1"] = 0
    > return "triangle_1"
```
This is causing issue, because there are two nodes named "triangle_1"

In the new implementation, here is the process:
```
[1] sanitizedName = uniqueName = "triangle"
    sanitizedName in this.nodeNamesUsed = False
    this.nodeNamesUsed["triangle"] = 0
    > return "triangle" 

[2] sanitizedName = uniqueName = "triangle"
    sanitizedName in this.nodeNamesUsed = True
    uniqueName = "triangle_1"
    this.nodeNamesUsed["triangle"] = 1
    this.nodeNamesUsed["triangle_1"] = 0
    > return "triangle_1"

[3] sanitizedName = uniqueName = "triangle_1"
    sanitizedName in this.nodeNamesUsed = True
    uniqueName = "triangle_1_1"
    this.nodeNamesUsed["triangle_1"] = 1
    this.nodeNamesUsed["triangle_1_1"] = 0
    > return "triangle_1_1"
```
This function will always create unique name for each node.

Hope this answer your question.
Let me know if you have any other concern.

Thank you.

Reply:
Ahh I see, thank you!

Reply:
Yeah, looks like ensuring the unique names, tho I haven't thought deeply yet.

In the example above, it would be good to respect the original names more ideally the result at [3] would be kept "triangle_1" and the result at [2] would be names something differently? If we want to do that, should we first look through all the original names? Anyways, we can improve later if needed. Thanks, good work.